dcrec/secp256k1/signature.go

```diff
@@ -93,7 +95,80 @@ func (sig *Signature) Serialize() []byte {
 // Verify returns whether or not the signature is valid for the provided hash
 // and secp256k1 public key.
 func (sig *Signature) Verify(hash []byte, pubKey *PublicKey) bool {
-	return ecdsa.Verify(pubKey.ToECDSA(), hash, sig.R, sig.S)
+	// The algorithm for verifying an ECDSA signature is given as algorithm 4.30
+	// in [GECC].
+	//
+	// The following is a paraphrased version for reference:
+	//
+	// G = curve generator
+	// N = curve order
+	// Q = public key
+	// m = message
+	// R, S = signature
+	//
+	// 1. Fail if R and S are not in [1, N-1]
+	// 2. e = H(m)
+	// 3. w = S^-1 mod N
+	// 4. u1 = e * w mod N
+	//    u2 = R * w mod N
+	// 5. X = u1G + u2Q
+	// 6. Fail if X is the point at infinity
+	// 7. x = X.x mod N (X.x is the x coordinate of X)
+	// 8. Verified if x == r
```

Comment:
I think this should be `x == R` (capital R instead of lower case r).

Reply:
Right. It's `r` in the book, but I agree it should match the rest of the comments which I already adapted accordingly.  Updated.